### PR TITLE
Remove some vis and vec fields from Policy

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/policy/policy.py [hunk 1]
```diff
@@ -32,23 +32,18 @@ def __init__(
...
-            if len(sen_spec.shape) == 1
-        )
-        self.vis_obs_size = sum(
-            1 for sen_spec in behavior_spec.sensor_specs if len(sen_spec.shape) == 3
-        )
-        self.use_continuous_act = self.behavior_spec.action_spec.is_continuous()
         self.previous_action_dict: Dict[str, np.ndarray] = {}
         self.memory_dict: Dict[str, np.ndarray] = {}
         self.normalize = trainer_settings.network_settings.normalize
+        if self.normalize:
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/policy/policy.py:38]: What happens if self.normalize is True but none of them are vector obs? I.e. can we just remove this flow?

File: ml-agents/mlagents/trainers/policy/policy.py [hunk 2]
```diff
@@ -32,23 +32,18 @@ def __init__(
...
+            has_vec_obs = False
```
Comment (andrewcoh) [inline ml-agents/mlagents/trainers/policy/policy.py:39]:
```suggestion
           self.normalize = False
```

File: ml-agents/mlagents/trainers/policy/policy.py [hunk 3]
```diff
@@ -32,23 +32,18 @@ def __init__(
...
+            # Make sure there is at least one vector observation for normalization
+            for sen_spec in behavior_spec.sensor_specs:
+                if len(sen_spec.shape) == 1:
+                    has_vec_obs = True
+                    break
+            if not has_vec_obs:
+                self.normalize = False
```
Comment (andrewcoh) [inline ml-agents/mlagents/trainers/policy/policy.py:46]:
```suggestion
            self.normalize = False
            # Make sure there is at least one vector observation for normalization
            for sen_spec in behavior_spec.sensor_specs:
                if len(sen_spec.shape) == 1:
                    self.normalize = True
                    break
```

Comment (andrewcoh) [reply ml-agents/mlagents/trainers/policy/policy.py:46]: :trollface:

File: ml-agents/mlagents/trainers/policy/policy.py (hunk 1 above)
Comment (vincentpierre) [reply ml-agents/mlagents/trainers/policy/policy.py:38]: We can remove it, I just thought this was cleaner, but it does add some complexity.

Comment (vincentpierre) [reply ml-agents/mlagents/trainers/policy/policy.py]: Removed

**Review by andrewcoh — APPROVED** [review]

**Pull request merged**